### PR TITLE
Monica/search bar crash fix

### PR DESCRIPTION
**Pull request by myo3**
Austin & my changes

### CODE REVIEW

File: TCAT/Cells/RouteTableViewCell.swift [hunk 1]
```diff
@@ -136,12 +133,12 @@ class RouteTableViewCell: UITableViewCell {
     // MARK: Style
     
     private func styleTravelTime(){
-        travelTimeLabel.font = UIFont(name: "SFUIText-Medium", size: 14.0)
+        travelTimeLabel.font = UIFont(name: FontNames.SanFrancisco.Medium, size: 14.0)
```
Comment (mattbarker016) [inline TCAT/Cells/RouteTableViewCell.swift:136]: Neat!

File: TCAT/Controllers/HomeViewController.swift [hunk 2]
```diff
@@ -38,6 +37,9 @@ class HomeViewController: UIViewController, UITableViewDelegate, UITableViewData
             tableView.reloadData()
         }
     }
+    let reachability = Reachability(hostname: "10.132.7.249")
```
Comment (mattbarker016) [inline TCAT/Controllers/HomeViewController.swift]: @AAAstorga Set default IP inside Reachability file

File: TCAT/Controllers/HomeViewController.swift [hunk 3]
```diff
@@ -209,32 +246,62 @@ class HomeViewController: UIViewController, UITableViewDelegate, UITableViewData
...
+
+    func description(forEmptyDataSet scrollView: UIScrollView!) -> NSAttributedString! {
+        let title = isNetworkDown ? "No Network Connection" : "Location Not Found"
+        let attrs = [NSForegroundColorAttributeName: UIColor.mediumGrayColor]
+        return NSAttributedString(string: title, attributes: attrs)
+    }
+
+    func buttonTitle(forEmptyDataSet scrollView: UIScrollView!, for state: UIControlState) -> NSAttributedString! {
+        let buttonTitle = isNetworkDown ? "Try Again" : ""
+        let attrs = [NSForegroundColorAttributeName: UIColor.init(red: 0.0, green: 118.0 / 255.0, blue: 255.0 / 255.0, alpha: 1.0)]
```
Comment (mattbarker016) [inline TCAT/Controllers/HomeViewController.swift]: Either use a color from the Shared file, or add this one

File: TCAT/Controllers/RouteOptionsViewController.swift [hunk 4]
```diff
@@ -478,6 +486,43 @@ CLLocationManagerDelegate {
...
+    }
+
+    func title(forEmptyDataSet scrollView: UIScrollView!) -> NSAttributedString! {
+        let message = currentlySearching ? "Looking for routes..." : "No Routes Found"
+        let attrs = [NSFontAttributeName: UIFont(name: FontNames.SanFrancisco.Regular, size: 14.0), NSForegroundColorAttributeName: UIColor.mediumGrayColor]
+        
+        return NSAttributedString(string: message, attributes: attrs)
+    }
+    
+//    func customView(forEmptyDataSet scrollView: UIScrollView!) -> UIView! {
```
Comment (mattbarker016) [inline TCAT/Controllers/RouteOptionsViewController.swift]: get rid

File: TCAT/Model/Route.swift [hunk 5]
```diff
@@ -27,44 +28,50 @@ class Route: NSObject, JSONDecodable {
...
         self.arrivalTime = arrivalTime
         self.routeSummary = routeSummary
         self.directions = directions
-        self.path = path
-        self.travelDistance = travelDistance
         self.lastStopTime = lastStopTime
     }
     
+    static func getRoutesArray(fromJson json: JSON) -> [Route] {
+        if (json["success"]=="false") {
```
Comment (mattbarker016) [inline TCAT/Model/Route.swift]: use boolValue for JSON

File: TCAT/Model/Route.swift [hunk 6]
```diff
@@ -75,9 +82,40 @@ class Route: NSObject, JSONDecodable {
...
+            let direction = directions[index]
+            if direction.type == .walk {
+                calculateWalkingDirections(direction) { (path) in
+                    direction.path = path
+                }
+            }
+        }
+    }
+    
+    private func calculateWalkingDirections(_ direction: Direction, _ completion: @escaping ([CLLocationCoordinate2D]) -> Void) {
```
Comment (mattbarker016) [inline TCAT/Model/Route.swift:109]: might fit better in Network? Up to you

**Review by mattbarker016 — COMMENTED** [review]
@AAAstorga can help too with changes

**Comment by mattbarker016** [comment]
For merge conflicts, stick with what's already on master for the Model files, RouteDetailVC, Network (I think). For the .pbxproj, I think we should add both of our changes (although there might be some son stuff that can go, all I care about the testNew.json).

File: TCAT/Model/Route.swift (hunk 6 above)
Comment (myo3) [reply TCAT/Model/Route.swift:109]: i think ill keep in Route.swift for now

Comment (mattbarker016) [reply TCAT/Model/Route.swift:109]: Ok!

**Comment by mattbarker016** [comment]
You just need to resolve the conflicts above still I think, @myo3

**Comment by myo3** [comment]
Yeah I'm merging it rn

**Pull request merged**